contracts/src/AccountRegistry.sol

```diff
@@ -442,13 +571,20 @@ contract AccountRegistry is Initializable, EIP712Upgradeable, Ownable2StepUpgrad
         uint256[] calldata siblingNodes,
         uint256 nonce
     ) external virtual onlyProxy onlyInitialized {
-        _validateAuthenticatorAddressNotInUse(newAuthenticatorAddress);
-        require(
-            oldAuthenticatorAddress != newAuthenticatorAddress, "Old and new authenticator addresses cannot be the same"
-        );
-        require(newAuthenticatorAddress != address(0), "New authenticator address cannot be the zero address");
+        if (accountIndex == 0 || nextAccountIndex <= accountIndex) {
```

Comment:
is this extra gas worth it?

Reply:
leaving as is, not in common flows (only in update and recover functions)